core/annotations/annotations.go

```diff
@@ -28,16 +30,39 @@ const (
 type ID struct {
 	Kind Kind
 	Name string
+
+	// Parts is optional and is used to store additional information about the
+	// entity.
+	Parts any
```

Comment:
Could this instead be a `map[string]any`? At least then you are giving out _some_ indication at to what it should be - essentially keys and values.